tests/json-patch-tests/tests.json

```diff
@@ -429,6 +429,12 @@
        "doc": {"foo":"bar"},
        "patch": [{"op": "add", "path": "/FOO", "value": "BAR"}],
        "expected": {"foo": "bar", "FOO": "BAR"}
+    },
+
+    { "comment": "adding to subfield of a non-object should fail",
+      "doc": {"foo": {"bar": 1}},
+      "patch": [{"op": "add", "path": "/foo/bar/baz", "value": "5"}],
+      "error": "attempting to add to subfield of non-object"
```

Comment:
Looks good, but please add the test to `cjson-utils-tests.json` instead. You can also make a pull request with the same test [here](https://github.com/json-patch/json-patch-tests), then I will pull the change into cJSON.